src/brod_producer.erl

```diff
@@ -214,16 +236,25 @@ handle_info({msg, Pid, CorrId, #produce_response{} = R},
   Partition = State#state.partition, %% assert
   case brod_kafka:is_error(ErrorCode) of
     true ->
+      ErrorDesc = brod_kafka_error:desc(ErrorCode),
       error_logger:error_msg(
         "Error in produce response\n"
         "Topic: ~s\n"
         "Partition: ~B\n"
         "Offset: ~B\n"
-        "Error code: ~p",
-        [Topic, Partition, Offset, ErrorCode]),
-      {stop, {produce_error, {Topic, Partition, Offset, ErrorCode}}, State};
+        "Error: ~s",
+        [Topic, Partition, Offset, ErrorDesc]),
+      Error = {produce_response_error, Topic, Partition, Offset, ErrorCode, ErrorDesc},
+      case is_retriable(ErrorCode) of
+        true ->
+          {ok, NewState} = schedule_retry(State, Error),
+          {ok, NewBuffer} = brod_producer_buffer:nack(Buffer, CorrId),
+          {noreply, NewState#state{buffer = NewBuffer}};
+        false ->
+          {stop, Error, State}
+      end;
     false ->
-      NewBuffer = brod_producer_buffer:ack(Buffer, CorrId),
+      {ok, NewBuffer} = brod_producer_buffer:ack(Buffer, CorrId),
```

Comment:
i can't find a place where #state.retries is set back to 0.
should it be here ?